io/net/bridge.py

```diff
@@ -35,19 +35,20 @@ def check_failure(self, cmd):
             self.fail("Command %s failed" % cmd)
 
     def setUp(self):
-        self.host_interfaces = self.params.get("interfaces",
-                                               default="").split(" ")
-        if not self.host_interfaces:
-            self.cancel("User should specify host interface/s")
-
+        self.host_interfaces = []
+        local = LocalHost()
+        interfaces = netifaces.interfaces()
+        devices = self.params.get("interfaces", default="").split(" ")
+        for device in devices:
+            if device in interfaces:
+                self.host_interfaces.append(device)
+            elif local.validate_mac_addr(device):
+                self.host_interfaces.append(local.get_interface_by_hwaddr(device).name)
+            else:
+                self.cancel("Please check the network device")
```

Comment:
@abdhaleegit Because you are `cancel` the test in a loop, is that the correct one?

Reply:
I did not get your point

Reply:
basically if the input is wrong, then the test gets canceld by breaking the loop and tests exits